### PR TITLE
issues #27 #20 #34 #54 implemented with New Contributors Feature

### PR DESCRIPTION
**Pull request by awcoetzer**
Hi Micha, 👋

I added the contributors feature, Yay!.

I as well fixed & temporarily fixed the previous pull request. Please request that @will-s-205 take a look at the code again.

The previous pull request removed some functionality I had added onto. I have gone ahead and restored that and commented out the code that was added as it needs some rework in terms of responsive design, the implementation of toggling themes can be done in a simpler manner, placement of HTML elements, as well as inline JavaScript was added and not needed, I went ahead and placed it as an event listener. 

I also left a comment to explain how one can contact me for further assistance if they need help on how to implement it. 

Hope this helps

### CODE REVIEW

**Comment by michaella23** [comment]
@awcoetzer thanks for this awesome contribution! Will merge a bit later. Question: are you able to go into the closed pull requests and add comments/review there? I see your comments in the code that you have pushed, but I think I'd prefer for requests for changes to be made either in GitHub or in the collaborative-work channel on Discord rather than commented code. Let me know what you find out. Thanks!

File: assets/javascript/app.js [hunk 1]
```diff
@@ -1,54 +1,82 @@
...
+    greetingEl.textContent = `Good ${timeOfDay} ${timeOfDayEmote} bootcamp students!`
 }, 1000)
 
 
-const themeForDaytime = () => {
-    // the user is saving the date object to a variable
-    const date = new Date();
-    // then pulling the hours from the date object
-    const hours = date.getHours();
+/*
```
Comment (michaella23) [inline assets/javascript/app.js:37]: @will-s-205 can you see these comments here in the changed files of this pull request? Also, do you notice you are able to add comments? If you hover over any line of code, there might be a blue '+' that you can click to add comments.

File: assets/javascript/app.js [hunk 2]
```diff
@@ -1,54 +1,82 @@
...
+//     } else if (document.getElementById("theme-button").textContent == "LIGHT") {
+//         document.body.style.color = "silver";
+//         document.body.style.background = "var(--clr-maroon-overload-will-s-205)"; //DARK
+//         document.getElementById("theme-button").textContent = "DARK"; //DARK
+//     } else if (document.getElementById("theme-button").textContent == "DARK") {
+//         document.body.style.color = "var(--clr-paragraph)";
+//         document.body.style.background = "var(--clr-background-tint)"; // LIGHT
+//         document.getElementById("theme-button").textContent = "LIGHT"; // LIGHT
+//     }
+// }
```
Comment (will-s-205) [inline assets/javascript/app.js:77]: Hi @awcoetzer. Thanks for letting me know. I've been working on daytime greeting logic and change of theme features. Code have been pushed into production. Please do me a favor and do NOT comment it out until we have a solution and some collaborative work together. I hope you understand. Thanks.
cc @michaella23

**Review by will-s-205 — CHANGES_REQUESTED** [review]

**Review by will-s-205 — CHANGES_REQUESTED** [review]

File: assets/javascript/app.js (hunk 1 above)
Comment (will-s-205) [reply assets/javascript/app.js:37]: Yes. I did leave a comment or two down below. Thanks @michaella23 for helping us

**Comment by michaella23** [comment]
I am merging this pull request, and appreciate all the work that went in @awcoetzer. 🙂  I wonder how we can improve communication on this project... 🤔 Perhaps we can do some project planning in the Discord channel? We could also create Issues when there is a question or if we are going to change code someone else has written. As well, it seems anyone is able to comment on code in the 'Files Changed' tab of Pull Requests. Open to suggestions.

**Comment by awcoetzer** [comment]
Hi @will-s-205, I apologize if I have offended you, though there was no need to be, I offered to help you work through some issues I picked up in your code, though after this I think Ill just share my thoughts here.

The code wasn't confusing in the sense that I couldn't understand it, it was confusing in the sense that you have essentially limited yourself and hard coded certain values. The theme button you implemented relies on checking the textContent property of the element. Small changes, such as whitespace or uppercase/lowercase differences, can break the comparison. 
It would be better to use a separate variable to store and check the current theme state.

The logic to toggle the theme seems to be inconsistent. When the current theme is "LIGHT," it changes to the dark theme, but when the current theme is "DARK," it changes to the light theme. This inconsistency could lead to **_confusion_** and unexpected behavior.

The conditions for hours < 17 and for hours > 17 seem to imply that the theme should change at exactly 5 PM. However, this code will only execute the first condition if the current hour is before 5 PM and the second condition if the current hour is after 5 PM. It does not handle the case when the current hour is exactly 5 PM. 

Regarding your note on commenting out code, I would say the same for you when deleting code when it comes to your part, I had my reasons to comment your code out, because 1. I wanted to work alongside you but 2 and most importantly it broke the responsive design of the site, something you did not account for and I didn't expect you to.

@michaella23 I think after this I will call it quits

**Comment by will-s-205** [comment]
@awcoetzer I'm not offended. Thanks for your consideration. I was trying to build constructive dialogue. We need to come up with rules or something. How do we work together on the same project and etc. Thanks again for your feedback

**Pull request merged**